test/external.dependencies.test.js

```diff
@@ -0,0 +1,16 @@
+// Copyright 2019 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+const { config } = require('../config')
+const request = require('request')
+
+describe('Checks for documentation websites', () => {
+  test('Check TSLint plugin documentation website', async () => {
+    request(config.tslintProjectWebsite, async (err, response, body) => {
+      if (err) {
+        throw new Error(err)
+      }
+      expect(response.statusCode === 200 || response.statusCode === 201).toBeTruthy()
```

Comment:
I believe this checks that the GitHub README is present, whereas what I'm more concerned about is whether the sections of the README disappear. Can we try and load one of the reference links and make sure there's content in the DOM where we expect it?

This might be a lot of work and not provide sufficient value to be worth the effort?

Reply:
Maybe there is a way to do this.
The problem is that if you write you write a unit test for a specific "label" section of the website like this: 

![image](https://user-images.githubusercontent.com/16246778/56352061-1fbf3b80-61d7-11e9-9048-dd92341b1604.png)

the test passes.
I have to find a way how to check for the existing of a specific label on the website.

Also, because we have all rule names hardcoded in our configuration options for TSLint (in tslint/tslint.json, it will be possible to update our tslint/tslint.json file with a new rule and then our unit test will be updated too.

Reply:
I haven't written JS for about 11 years, but back then I would have used [`document.getElementById()`](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementById).

We can see from the HTML of tslint-config-security's README.md that the anchor's all have consistent id's following a pattern based on the target name, for example:
`<a id="user-content-tsr-detect-unsafe-regexp" ...>`
and
`<a id="user-content-tsr-detect-child-process" ...>`

Reply:
I added tests for that using a really great library for that called [cheerio](https://github.com/cheeriojs/cheerio).